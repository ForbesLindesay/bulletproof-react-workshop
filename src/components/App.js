// @flow

import React from 'react';
import AddStoryContainer from './AddStoryContainer';
import Spinner from './Spinner';
import Story from './Story';

type Props = {|
  +loading: boolean,
  +stories: Array<{|+id: string, +votes: number, +body: string|}>,
  +onVote: (id: string) => mixed,
  +onAddStory: (body: string) => mixed,
|};

export default function App(props: Props) {
  if (props.loading) return <Spinner />;
  return (
    <div>
      {props.stories.map(story => (
        <Story
          key={story.id}
          id={story.id}
          votes={story.votes}
          body={story.body}
          onVote={props.onVote}
        />
      ))}
      <AddStoryContainer onAddStory={props.onAddStory} />
    </div>
  );
}
