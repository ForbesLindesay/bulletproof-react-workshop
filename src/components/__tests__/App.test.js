// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

test('App', () => {
  expect(
    renderer
      .create(
        <App
          loading={true}
          stories={[]}
          onAddStory={() => {}}
          onVote={() => {}}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot('loading');

  expect(
    renderer
      .create(
        <App
          loading={false}
          stories={[]}
          onAddStory={() => {}}
          onVote={() => {}}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot('no stories');

  expect(
    renderer
      .create(
        <App
          loading={false}
          stories={[
            {id: 'a', body: 'Awesome story a', votes: 5},
            {id: 'b', body: 'Awesome story b', votes: 3},
          ]}
          onAddStory={() => {}}
          onVote={() => {}}
        />,
      )
      .toJSON(),
  ).toMatchSnapshot('with 2 stories');
});
