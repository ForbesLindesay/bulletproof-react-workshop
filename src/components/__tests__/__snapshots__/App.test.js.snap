// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading 1`] = `
<div>
  Loading...
</div>
`;

exports[`no stories 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      data-test-id="NewStoryBody"
      onChange={[Function]}
      value=""
    />
    <button
      data-test-id="AddStoryButton"
      type="submit"
    >
      Add Story
    </button>
  </form>
</div>
`;

exports[`with 2 stories 1`] = `
<div>
  <article
    data-test-id="story"
  >
    <span
      className="vote-count"
      data-test-id="VoteCount"
    >
      5
    </span>
    <div
      className="story-body"
      data-test-id="StoryBody"
    >
      Awesome story a
    </div>
    <button
      data-test-id="AddVote"
      onClick={[Function]}
      type="button"
    >
      Vote
    </button>
  </article>
  <article
    data-test-id="story"
  >
    <span
      className="vote-count"
      data-test-id="VoteCount"
    >
      3
    </span>
    <div
      className="story-body"
      data-test-id="StoryBody"
    >
      Awesome story b
    </div>
    <button
      data-test-id="AddVote"
      onClick={[Function]}
      type="button"
    >
      Vote
    </button>
  </article>
  <form
    onSubmit={[Function]}
  >
    <input
      data-test-id="NewStoryBody"
      onChange={[Function]}
      value=""
    />
    <button
      data-test-id="AddStoryButton"
      type="submit"
    >
      Add Story
    </button>
  </form>
</div>
`;
